src/components/Calendar/helpers/buildNewRangeFromValue.js

```diff
@@ -2,37 +2,34 @@ import isDateWithinRange from './isDateWithinRange';
 import compareDates from './compareDates';
 
 export default function buildNewRangeFromValue(value, currentRange, currentUpdatePosition = 0) {
-    if (!currentRange || currentRange.length === 0)
-        return {
-            range: [value],
-        };
+    if (currentRange && currentRange.length > 0) {
+        const [rangeStart, rangeEnd] = currentRange;
+        rangeStart.setHours(0, 0, 0, 0);
```

Comment:
be carefull doing this since these date methods make mutations and can lead to unexpected behaviors and hard to track